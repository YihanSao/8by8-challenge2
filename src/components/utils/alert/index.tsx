export { Alert } from './alert';
export { useAlert } from './use-alert';
