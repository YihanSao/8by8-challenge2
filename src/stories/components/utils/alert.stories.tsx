import { Meta, StoryObj } from '@storybook/react';
import { Alert, useAlert } from '@/components/utils/alert';
import { GlobalStylesProvider } from '@/stories/global-styles-provider';

const meta: Meta<typeof Alert> = {
  component: Alert,
};

export default meta;

type Story = StoryObj<typeof Alert>;

function ShowAlerts() {
  const { alertRef, showAlert } = useAlert();

  const showErrorAlert = () => {
    showAlert('Your error message here.', 'error');
  };

  const showSuccessAlert = () => {
    showAlert('Your success message here.', 'success');
  };

  return (
    <GlobalStylesProvider>
      <Alert ref={alertRef} />
      <button onClick={showErrorAlert}>Show Error Alert</button>
      <button onClick={showSuccessAlert}>Show Success Alert</button>
    </GlobalStylesProvider>
  );
}

export const Default: Story = {
  render: ShowAlerts,
};
