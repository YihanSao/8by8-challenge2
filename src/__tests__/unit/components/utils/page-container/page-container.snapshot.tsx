import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react';
import { PageContainer } from '@/components/utils/page-container';

describe('PageContainer', () => {
  afterEach(cleanup);
  it('renders the PageContainer unchanged.', () => {
    const { container } = render(<PageContainer />);
    expect(container).toMatchSnapshot();
  });
});
