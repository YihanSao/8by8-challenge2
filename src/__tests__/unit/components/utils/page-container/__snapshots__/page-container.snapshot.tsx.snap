// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer renders the PageContainer unchanged. 1`] = `
<div>
  <div
    class="outer_container"
  >
    <div
      class="inner_container"
    />
  </div>
</div>
`;
